### PR TITLE
feat(docs): output docs from JSDoc

### PR DESCRIPTION
**Pull request by kennylam**
### Related Ticket(s)

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5635

### Description

This adds documentation using JSDoc. Documentation is automatically generated and and pushed to a `docs` branch each time a pull request is merged into `main`. The `docs` branch is used for Github pages.

### Changelog

**New**

- JSDoc documentation
- action to deploy JSDoc files to Github Pages

**Changed**

- update JSDoc annotations in gather/audit files
- update `.gitignore`

### CODE REVIEW

File: src/config/custom-config.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ module.exports = {
     onlyCategories: [
       'legal',
       'page-data',
-      'page-content',
+      // 'page-content',
```
Comment (jeffchew) [inline src/config/custom-config.js:52]: Should this be removed?

**Review by jeffchew — APPROVED** [review]
Minor comment, otherwise looks good to me!

File: src/config/custom-config.js (hunk 1 above)
Comment (kennylam) [reply src/config/custom-config.js:52]: @jeffchew That refers to a category that is commented out. Beacon would complain that `page-content` didn't exist so I commented it out for now. Not sure if it's safe to remove @IgnacioBecerra? Or is this something you were still working on. I missed it in the last PR review.

https://github.com/IBM/beacon-for-ibm-dotcom/blob/8bf51d72de6b7deb2a2bbcc0a89c7defecd8102c/src/config/custom-config.js#L200-L205

Comment (IgnacioBecerra) [reply src/config/custom-config.js:52]: It should be safe to remove, I think this one was a placeholder audit that was never created (as far as I know).

**Pull request merged**